tony-portal/app/history/HistoryFileMover.java

```diff
@@ -99,4 +151,65 @@ private void moveIntermediateToFinished(FileSystem fs, FileStatus[] jobDirs, Zon
   private boolean jobInProgress(String jhistFileName) {
     return !jhistFileName.endsWith(Constants.HISTFILE_SUFFIX);
   }
+
+  // This function queries and returns Yarn for the App IDs of apps with YarnApplicationState.KILLED
+
+  private List<String> getKilledApps() throws IOException, YarnException {
+    List<String> totalKilledAppIds = new ArrayList<String>();
+    long currentTimestamp = System.currentTimeMillis();
+
+    //set timeDelta for 36 hours
+    long timeDelta = 36 * 60 * 60 * 1000;
+    YarnConfiguration yarnConf = new YarnConfiguration();
+    if (System.getenv(Constants.HADOOP_CONF_DIR) != null) {
+      yarnConf.addResource(new Path(System.getenv(Constants.HADOOP_CONF_DIR) +
+          File.separatorChar + Constants.CORE_SITE_CONF));
+      yarnConf.addResource(new Path(System.getenv(Constants.HADOOP_CONF_DIR) +
+          File.separatorChar + Constants.YARN_SITE_CONF));
+    }
+    YarnClient yarnClient = YarnClient.createYarnClient();
+    yarnClient.init(yarnConf);
+    yarnClient.start();
+    List<ApplicationReport> totalKilledApps = yarnClient.getApplications(EnumSet.of(YarnApplicationState.KILLED));
+
+    // Get apps within a specified time period. This prevents the History File mover from iterating through
+    //previously checked killed app IDs upon every iteration.
+
+    List<String> timeDeltaKilledIds = totalKilledApps
+        .stream()
+        .filter(x -> x.getStartTime() > (currentTimestamp - timeDelta))
+        .map(x -> x.getApplicationId().toString())
+        .collect(Collectors.toList());
+
+    LOG.info("Number Killed Yarn Apps within timeDelta: " + timeDeltaKilledIds.size());
+    return timeDeltaKilledIds;
+  }
+
+  private void renameKilledApps(FileSystem fs, List<FileStatus> jobDirs) {
+
+    for (FileStatus jobDir : jobDirs) {
+      String jhistFilePath = ParserUtils.getJhistFilePath(fs, jobDir.getPath());
+
+      if (jhistFilePath.endsWith(".jhist.inprogress")) {
+        {
```

Comment:
I think we can remove this set of braces?

Reply:
Added this as a safeguard against potential errant appending of the filename due to system anomalies.

Reply:
Jonathan meant this "{" in line 161, I don't that is needed

Reply:
Yeah, just the braces, not the stuff in between